core/engine/src/bytecompiler/mod.rs

```diff
@@ -1737,6 +1754,17 @@ impl<'ctx> ByteCompiler<'ctx> {
         }
         self.r#return(false);
 
+        let mapped_arguments_binding_indices = if self.emitted_mapped_arguments_object_opcode {
+            MappedArguments::binding_indices(&self.params)
+        } else {
+            ThinVec::new()
+        };
```

Comment:
```suggestion
        let mapped_arguments_binding_indices = self.emitted_mapped_arguments_object_opcode
        .then(|| MappedArguments::binding_indices(&self.params))
        .unwrap_or_default();
```
We can simplify by using [`bool::then()`](https://doc.rust-lang.org/std/primitive.bool.html#method.then)